simulator/infra/replacement/cache_replacement.cpp

```diff
@@ -68,10 +69,131 @@ std::size_t LRUCacheInfo::update()
     return lru_elem;
 }
 
+//////////////////////////////////////////////////////////////////
+
+struct node
+{
+    node(int new_value = 0, std::size_t new_pointer = 0)
+    : value( new_value), pointer( new_pointer) {}
+
+    int value;
+    std::size_t pointer;
```

Comment:
The names may be more descriptive here. I have not read the rest of the code yet, so I do not understand what is the difference between value and pointer, why both are needed etc.

Reply:
Ok, I'm back here.
Should `pointer` be a Left/Right switch?

Reply:
Hope it is more clear now